enterprise_access/apps/bffs/context.py

```diff
@@ -40,6 +43,23 @@ def request(self):
     def user(self):
         return self._request.user
 
+    def initialize_common_context_data(self):
+        """
+        Initialize commonly used context attributes, such as enterprise customer UUID and LMS user ID.
+        """
+        if request_object := (getattr(self.request, 'query_params', {}) or getattr(self.request, 'data', None)):
+            enterprise_customer_uuid = request_object.get('enterprise_customer_uuid')
+            if enterprise_customer_uuid:
+                self.enterprise_customer_uuid = enterprise_customer_uuid
+            else:
+                raise KeyError("enterprise_customer_uuid is required for this request.")
+        else:
+            raise ValueError("A query parameter or data object is required for this request.")
```

Comment:
Generally, `query_params` will always exist in practice (outside of mocked test requests); `data` will only exist for POST requests. This `else` considers the case where neither is present, which I don't think can happen in practice.

Might be best to ensure the test more accurately reflects actual usage by making a POST call when `data` is truthy, and perhaps updating this method to account for `self.request.method === 'POST`, e.g.:

```python
def initialize_common_context_data(self):
        """
        Initialize commonly used context attributes, such as enterprise customer UUID and LMS user ID.
        """
        # Set enterprise_customer_uuid from request parameters or previously set context
        enterprise_uuid_query_param = self.request.query_params.get('enterprise_customer_uuid')
        enterprise_uuid_post_param = None
        if self.request.method == 'POST':
            enterprise_uuid_post_param = self.request.data.get('enterprise_customer_uuid')

        enterprise_customer_uuid = enterprise_uuid_query_param or enterprise_uuid_post_param
        if enterprise_customer_uuid:
            self.enterprise_customer_uuid = enterprise_customer_uuid
        else:
            raise ValueError("enterprise_customer_uuid is required for this request.")

        # Set lms_user_id from the authenticated user object in the request
        self.lms_user_id = self.user.lms_user_id
```

```python
@ddt.data(
        {
            'query_params': True,
            'data': True,
        },
        {
            'query_params': False,
            'data': True,
        },
        {
            'query_params': True,
            'data': False,
        },
        {
            'query_params': False,
            'data': False,
        },
    )
    @ddt.unpack
    def test_handler_context_enterprise_customer_uuid_param(self, query_params, data):
        if not query_params:
            self.request.query_params = {}

        if data:
            self.request = self.factory.post('sample/api/call')
            self.request.data = {
                'enterprise_customer_uuid': self.mock_enterprise_customer_uuid
            }

        if not (query_params or data):
            with self.assertRaises(ValueError):
                HandlerContext(self.request)
        else:
            self.assertEqual(self.context.enterprise_customer_uuid, self.mock_enterprise_customer_uuid)
            self.assertEqual(self.context.lms_user_id, self.mock_user.lms_user_id)
```

I feel this might be preferable over adding complexity to the implementation to appease mock test data? 🤔

Reply:
Updated 👍🏽